File: frontend/src/types/invoice.ts
export interface Invoice {
  id: string
  invoice_no: string | null
  invoice_date: string | null
  invoice_type: string | null
  seller_name: string | null
  seller_tax_no: string | null
  amount: number
  tax_amount: number
  total_amount: number
  expense_category: string | null
  reimbursement_person: string | null
  confidence: number
  anomaly_flag: string | null
  anomaly_reason: string | null
  image_path: string | null
  created_at: string
  updated_at: string
}

export interface InvoiceListResponse {
  items: Invoice[]
  total: number
  page: number
  size: number
}

export interface CategorySummary {
  category: string
  count: number
  amount: number
  tax_amount: number
}

export interface SummaryResponse {
  by_category: CategorySummary[]
  total_count: number
  total_amount: number
  total_tax: number
  anomaly_count: number
}

export interface UploadResponse {
  task_id: string
  total_count: number
  processed: number
  message: string
}

export interface VoucherEntry {
  编制日期: string
  凭证类型: string
  凭证序号: number
  凭证号: string
  制单人: string
  附件张数: number
  会计年度: string
  科目编码: string
  科目名称: string
  凭证摘要: string
  借贷方向: string
  金额: number
  币种: string
  汇率: number
  原币金额: number
  数量?: number | null
  单价?: number | null
  结算方式名称?: string | null
  结算日期?: string | null
  结算票号?: string | null
  业务日期?: string | null
  员工编号?: string | null
  员工姓名?: string | null
  往来单位编号?: string | null
  往来单位名称?: string | null
  货品编号?: string | null
  货品名称?: string | null
  部门名称?: string | null
  项目名称?: string | null
}

export interface VoucherGenerateResponse {
  vouchers: VoucherEntry[]
  total_debit: number
  total_credit: number
}

File: backend/app/schemas/invoice.py
from datetime import date, datetime
from typing import Optional, List, Any
from pydantic import BaseModel


class InvoiceBase(BaseModel):
    invoice_no: Optional[str] = None
    invoice_date: Optional[date] = None
    invoice_type: Optional[str] = None
    seller_name: Optional[str] = None
    seller_tax_no: Optional[str] = None
    amount: float = 0
    tax_amount: float = 0
    total_amount: float = 0
    expense_category: Optional[str] = None
    reimbursement_person: Optional[str] = None
    confidence: float = 0
    anomaly_flag: Optional[str] = None
    anomaly_reason: Optional[str] = None


class InvoiceCreate(InvoiceBase):
    image_path: Optional[str] = None
    raw_response: Optional[dict] = None


class InvoiceUpdate(BaseModel):
    expense_category: Optional[str] = None
    reimbursement_person: Optional[str] = None
    anomaly_flag: Optional[str] = None
    anomaly_reason: Optional[str] = None


class InvoiceResponse(InvoiceBase):
    id: str
    image_path: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class InvoiceListResponse(BaseModel):
    items: List[InvoiceResponse]
    total: int
    page: int
    size: int


class CategorySummary(BaseModel):
    category: str
    count: int
    amount: float
    tax_amount: float


class SummaryResponse(BaseModel):
    by_category: List[CategorySummary]
    total_count: int
    total_amount: float
    total_tax: float
    anomaly_count: int


class UploadResponse(BaseModel):
    task_id: str
    total_count: int
    processed: int
    message: str


class VoucherEntry(BaseModel):
    编制日期: str
    凭证类型: str
    凭证序号: int
    凭证号: str
    制单人: str
    附件张数: int
    会计年度: str
    科目编码: str
    科目名称: str
    凭证摘要: str
    借贷方向: str
    金额: float
    币种: str = "人民币"
    汇率: float = 1
    原币金额: float = 0
    数量: Optional[float] = None
    单价: Optional[float] = None
    结算方式名称: Optional[str] = None
    结算日期: Optional[str] = None
    结算票号: Optional[str] = None
    业务日期: Optional[str] = None
    员工编号: Optional[str] = None
    员工姓名: Optional[str] = None
    往来单位编号: Optional[str] = None
    往来单位名称: Optional[str] = None
    货品编号: Optional[str] = None
    货品名称: Optional[str] = None
    部门名称: Optional[str] = None
    项目名称: Optional[str] = None


class VoucherGenerateRequest(BaseModel):
    invoice_ids: List[str]
    voucher_date: str
    voucher_type: str = "转"
    maker: str = "系统"
    department: str = ""


class VoucherGenerateResponse(BaseModel):
    vouchers: List[VoucherEntry]
    total_debit: float
    total_credit: float
